backend/dataall/modules/dataset_sharing/services/share_managers/s3_bucket_share_manager.py

```diff
@@ -602,6 +626,7 @@ def generate_enable_pivot_role_permissions_policy_statement(pivot_role_name, sou
                 'kms:ReEncrypt*',
                 'kms:TagResource',
                 'kms:UntagResource',
+                'kms:DescribeKey',
```

Comment:
The pivot role IAM policies include:
- "kms:List*", to ALL resources
- "kms:DescribeKey", to ALL resources

And then for data.all-created keys + imported keys:
```
            "Action": [
                "kms:Decrypt",
                "kms:Encrypt",
                "kms:GenerateDataKey*",
                "kms:GetKeyPolicy",
                "kms:PutKeyPolicy",
                "kms:ReEncrypt*",
                "kms:TagResource",
                "kms:UntagResource"
```

I think we should include the same permissions in the KMS resource policy: in s3_bucket_share_manager, in s3_access_points_manager and in the dataset_stack!

Reply:
even better we define this once as a constant somewhere in dataset module and reference the permissions block in all 3 locations

Reply:
I will have a look at it, but it can only be placed in `datasets_base` because otherwise it creates a circular dependency. Can I make a comment about this and solve it in the design of #1123 ?